pivot.py

```diff
@@ -0,0 +1,1723 @@
+            main_body += flash_div
+            main_body += self.main
+            #main_body += Footer().tag()
+        return html_layout
+
+class Index(Component):
+    'Index'
+    __name__ = 'www.index'
+    _path = None
+
+    database_name = fields.Char('Database Name')
+    table_name = fields.Char('Table Name')
+    table_properties = fields.Char('Table Properties')
+
+    @classmethod
+    def get_url_map(cls):
+        return [
+            Rule('/<string:database_name>/babi/pivot/<string:table_name>/<string:table_properties>')
+        ]
+
+    def render(self):
+        pool = Pool()
+        Layout = pool.get('www.layout')
+        Pivot = pool.get('www.pivot_table')
+        PivotHeader = pool.get('www.pivot_header')
+        BabiTable = pool.get('babi.table')
+        User = pool.get('res.user')
+        Index = pool.get('www.index')
+
+        # Get the table name
+        # If table name starts with "__" we need to remove this part to search the table name
+        table_name = self.table_name
+        if table_name.startswith('__'):
+            table_name = table_name.split('__')[-1]
+
+        babi_tables = BabiTable.search([
+            ('internal_name', '=', table_name)], limit=1)
+        if babi_tables:
+            babi_table = babi_tables[0]
+            table_name = babi_table.name
+
+            users = User.search([('id', '=', Transaction().user)], limit=1)
+            user = users[0] if users else None
+            error_page = False
+            if not user:
+                error_page = True
+            if not error_page and user and babi_table.access_users:
+                if user not in babi_table.access_users:
+                    error_page = True
+            if not error_page and user and babi_table.access_groups:
+                error_page = True
+                for group in user.groups:
+                    if group in babi_table.access_groups:
+                        error_page = False
+                        break
+
+            if error_page:
+                with main(cls="grid min-h-full place-items-center bg-white px-6 py-24 sm:py-32 lg:px-8") as error_section:
+                    with div(cls="text-center"):
+                        p('404', cls="text-base font-semibold text-indigo-600")
+                        h1('Page not found', cls="mt-4 text-3xl font-bold tracking-tight text-gray-900 sm:text-5xl")
+                        p('Sorry, we couldn’t find the page you’re looking for.', cls="mt-6 text-base leading-7 text-gray-600")
+                layout = Layout(title=f'Page not found | Tryton')
+                layout.main.add(error_section)
+                return layout.tag()
+
+        # We cant sent an empty value, the routes dont detect correctly the
+        # field and fails
+        if not hasattr(self, 'table_properties'):
+            table_properties = 'null'
+        else:
+            table_properties = self.table_properties
```

Comment:
You say routes d'on detect correctly but that is probably because the alternative routes are not defined.

If you added:

```
Rule('/<string:database_name>/babi/pivot/<string:table_name>/'),
Rule('/<string:database_name>/babi/pivot/<string:table_name>'),
```
it would work.